processing/src/main/java/org/apache/druid/segment/DimensionHandlerUtils.java

```diff
@@ -358,58 +414,111 @@ public static Float convertObjectToFloat(@Nullable Object valObj)
+        if (fieldName != null) {
+          message = StringUtils.nonStrictFormat(
+              "Could not ingest value [%s] as float for dimension [%s]. A float column cannot have multiple values in the same row.",
+              valObj,
+              fieldName
+          );
+        } else {
+          message = StringUtils.nonStrictFormat(
+              "Could not ingest value [%s] as float. A float column cannot have multiple values in the same row.",
+              valObj
+          );
+        }
+        throw new ParseException(
+            valObj.getClass().toString(),
+            message
+        );
+      } else {
+        final String message;
+        if (fieldName != null) {
+          message = StringUtils.nonStrictFormat(
+              "Could not convert value [%s] to float for dimension [%s]. Invalid type: [%s]",
+              valObj,
+              fieldName,
+              valObj.getClass()
+          );
+        } else {
+          message = StringUtils.nonStrictFormat(
+              "Could not convert value [%s] to float. Invalid type: [%s]",
+              valObj,
+              valObj.getClass()
+          );
+        }
+        throw new ParseException(
+            valObj.getClass().toString(),
+            message
+        );
       }
-      return ret;
-    } else if (valObj instanceof List) {
-      throw new ParseException(
-          valObj.getClass().toString(),
-          "Could not ingest value %s as float. A float column cannot have multiple values in the same row.",
-          valObj
-      );
-    } else {
-      throw new ParseException(
-          valObj.getClass().toString(),
-          "Could not convert value [%s] to float. Invalid type: [%s]",
-          valObj,
-          valObj.getClass()
-      );
     }
   }
 
   @Nullable
   public static Object convertObjectToType(
       @Nullable final Object obj,
       final TypeSignature<ValueType> type,
-      final boolean reportParseExceptions
+      final boolean reportParseExceptions,
+      @Nullable final String fieldName
   )
   {
     Preconditions.checkNotNull(type, "type");
 
     switch (type.getType()) {
       case LONG:
-        return convertObjectToLong(obj, reportParseExceptions);
+        return convertObjectToLong(obj, reportParseExceptions, fieldName);
       case FLOAT:
-        return convertObjectToFloat(obj, reportParseExceptions);
+        return convertObjectToFloat(obj, reportParseExceptions, fieldName);
       case DOUBLE:
-        return convertObjectToDouble(obj, reportParseExceptions);
+        return convertObjectToDouble(obj, reportParseExceptions, fieldName);
       case STRING:
         return convertObjectToString(obj);
       case ARRAY:
         return coerceToObjectArrayWithElementCoercionFunction(
             obj,
-            x -> DimensionHandlerUtils.convertObjectToType(x, type.getElementType())
+            x -> DimensionHandlerUtils.convertObjectToType(x, type.getElementType(), false, fieldName)
```

Comment:
this should probably pass `reportParseExceptions` instead of `false`?

Reply:
yeah. I set it to `false` to preserve how it was working before. but it wouldn't hurt to have it pass `reportParseExceptions` instead